Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,376 @@
+
+
+class TestBoostingManagerSysDir(TestBoostingManagerSys):
+    """
+    testing class for directory source
+    """
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerSysDir, self).__init__(*argv, **kwargs)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSysDir, self).setUp()
+
+    def set_boosting_settings(self):
+        super(TestBoostingManagerSysDir, self).set_boosting_settings()
+        # self.bsettings.source_interval = 3
+        # self.bsettings.swarm_interval = 3
+        # self.bsettings.initial_swarm_interval = 10
+
+    def test_dir(self):
+        """
+        test directory filled with .torrents
+        """
+        self.boosting_manager.add_source(TESTS_DATA_DIR)
+
+        len_source = len(self.boosting_manager.boosting_sources)
+
+        # deliberately try to add the same source
+        self.boosting_manager.add_source(TESTS_DATA_DIR)
+        self.assertEqual(len(self.boosting_manager.boosting_sources), len_source, "identical source added")
+
+        dir_obj = self.boosting_manager.get_source_object(TESTS_DATA_DIR)
+        self.assertTrue(dir_obj.ready, "Not Ready")
+
+        count = 0
+        while len(dir_obj.torrents) < 2:
+            time.sleep(3)
+            count += 1
+
+        self.assertGreater(len(dir_obj.torrents), 2, "not ready torrent")
+
+        dir_obj.kill_tasks()
+
+        self.assertFalse(dir_obj.ready, "object still there")
+
+    def test_dir_archive_example(self):
+        """
+        test archive mode. Use diretory because easier to fetch torrent
+        """
+        self.boosting_manager.add_source(TESTS_DATA_DIR)
+        self.boosting_manager.set_archive(TESTS_DATA_DIR, True)
+
+        dir_obj = self.boosting_manager.get_source_object(TESTS_DATA_DIR)
+        self.assertTrue(dir_obj.ready, "Not Ready")
+
+        count = 0
+        while len(dir_obj.torrents) < 2:
+            time.sleep(3)
+            count += 1
+
+        self.assertGreater(len(dir_obj.torrents), 2, "not ready torrent")
+
+
+
+        dir_obj.kill_tasks()
+
+        self.assertFalse(dir_obj.ready, "object still there")
+
+    def tearDown(self):
+        super(TestBoostingManagerSysDir, self).tearDown()
+
+
+class TestBoostingManagerSysChannel(TestBoostingManagerSys):
+    """
+    testing class for channel source
+    """
+
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerSysChannel, self).__init__(*argv, **kwargs)
+        self.tdef = TorrentDef.load(TORRENT_FILE)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSysChannel, self).setUp()
+        self.channel_db_handler = self.session.open_dbhandler(NTFY_CHANNELCAST)
+        self.channel_db_handler._get_my_dispersy_cid = lambda: "myfakedispersyid"
+        self.votecast_db_handler = self.session.open_dbhandler(NTFY_VOTECAST)
+
+    def insert_channel_in_db(self, dispersy_cid, peer_id, name, description):
```

Comment:
These methods are also part of my test suite of the RESTApi. Maybe we should discuss moving these to another location (i.e. `TriblerCoreTest`)?